lsp-mode.el

```diff
@@ -5646,29 +5659,35 @@ perform the request synchronously."
            (lsp-request "workspace/symbol" `(:query ,pattern))))
 
 (defun lsp--get-symbol-to-rename ()
-  "Get symbol to rename and placeholder at point."
-  (if (let ((rename-provider (or (lsp--capability :renameProvider)
-                                 (-some-> (lsp--registered-capability "textDocument/rename")
-                                   (lsp--registered-capability-options)))))
-        (lsp:rename-options-prepare-provider? rename-provider))
-      (-when-let (response (lsp-request "textDocument/prepareRename"
-                                        (lsp--text-document-position-params)))
-        (-let* (((start . end) (lsp--range-to-region
-                                (if (lsp-range? response)
-                                    response
-                                  (lsp:prepare-rename-result-range response))))
-                (symbol (buffer-substring-no-properties start end))
-                (placeholder (lsp:prepare-rename-result-placeholder response)))
-          (cons symbol (or placeholder symbol))))
-    (let ((symbol (thing-at-point 'symbol t)))
-      (cons symbol symbol))))
+  "Get symbol to rename and placeholder at point.
+Returns a cons (SYMBOL-TO-RENAME . PLACEHOLDER)."
+  ;; Is there some connected server that supports renaming?
+  (if-let (lsp-feature? "textDocument/rename")
+      ;; Do we have prepare rename? Do it in that case. If we fail, don't do a
+      ;; rename, since it's invalid.
+      (if (lsp-feature? "textDocument/prepareRename")
+          (when-let ((prepare
+                      (lsp-request "textDocument/prepareRename"
+                                   (lsp--text-document-position-params))))
+            (-let* (((start . end) (lsp--range-to-region
+                                    (if (lsp-range? prepare)
+                                        prepare
+                                      (lsp:prepare-rename-result-range prepare))))
+                    (symbol (buffer-substring start end))
+                    (placeholder (lsp:prepare-rename-result-placeholder prepare)))
+              (cons symbol (or placeholder symbol))))
+        ;; We don't support prepare-rename, so just yield the symbol at point.
+        (let ((sym (thing-at-point 'symbol)))
+          (cons sym sym)))
+    ;; No connected server supports renaming.
```

Comment:
can you avoid putting comments like that? This should be evident from the content of the next line.

Reply:
I usually don't do that - this one was an accident. Fixed it.